### ISSUE TITLE
Husky dep problem

### ISSUE THREAD

**Issue by ORESoftware** [typicode/husky#109]
I got this error when running NPM install on a prod server:
```

remote: npm ERR! code ELIFECYCLE
remote: 
remote: npm ERR! husky@0.13.2 install: `node ./bin/install.js`
remote: npm ERR! Exit status 8
remote: npm ERR! 
remote: npm ERR! Failed at the husky@0.13.2 install script 'node ./bin/install.js'.
remote: npm ERR! This is most likely a problem with the husky package,
remote: npm ERR! not with npm itself.
remote: npm ERR! Tell the author that this fails on your system:
remote: npm ERR!     node ./bin/install.js
remote: npm ERR! You can get their info via:
remote: npm ERR!     npm owner ls husky
remote: npm ERR! There is likely additional logging output above.
remote: npm info preuninstall husky@0.13.2
remote: npm info uninstall husky@0.13.2
remote: 
remote: > husky@0.13.2 uninstall /var/lib/openshift/emsa-k76pvx3istg-1/app-root/runtime/repo/node_modules/csv2json/node_modules/husky
remote: > node ./bin/uninstall.js
remote: 
remote: 
remote: module.js:340
remote:     throw err;
remote:           ^
remote: Error: Cannot find module 'ansi-regex'
remote:     at Function.Module._resolveFilename (module.js:338:15)
remote:     at Function.Module._load (module.js:280:25)
remote:     at Module.require (module.js:364:17)
remote:     at require (module.js:380:17)
remote:     at Object.<anonymous> (/var/lib/openshift/emsa-k76pvx3istg-1/app-root/runtime/repo/node_modules/chalk/node_modules/strip-ansi/index.js:2:17)
remote:     at Module._compile (module.js:456:26)
remote:     at Object.Module._extensions..js (module.js:474:10)
remote:     at Module.load (module.js:356:32)
remote:     at Function.Module._load (module.js:312:12)
remote:     at Module.require (module.js:364:17)
remote: npm info husky@0.13.2 Failed to exec uninstall script
remote: npm WARN continuing anyway husky@0.13.2 uninstall: `node ./bin/uninstall.js`
remote: npm WARN continuing anyway Exit status 8
remote: npm info postuninstall husky@0.13.2
remote: npm info preuninstall csv2json@1.3.0
remote: npm info uninstall csv2json@1.3.0
remote: npm info postuninstall csv2json@1.3.0

```

not sure if it has to do with Husky or not but just in case

**Comment by danawoodman** [comment]
Similar issue here:

```shell
node_modules/strip-ansi/index.js:2
var ansiRegex = require('ansi-regex')();
                                     ^

TypeError: require(...) is not a function
    at Object.<anonymous> (/Users/dana/code/open-source/simple-sql-model/node_modules/strip-ansi/index.js:2:38)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/dana/code/open-source/simple-sql-model/node_modules/chalk/index.js:4:17)
    at Module._compile (module.js:571:32)
```

**Comment by ORESoftware** [comment]
yeah this seems bad Husky maintainers, can you take a look

**Comment by danawoodman** [comment]
@typicode any thoughts on this?

**Comment by typicode** [comment]
Hi,

Thank you for the details and sorry for the delay, I'll have a look.
What's your version of Node?

**Comment by danawoodman** [comment]
@typicode `v7.8.0`

**Comment by ORESoftware** [comment]
I am on Node v 6.9.5

**Comment by chrisui** [comment]
I'm getting the following error when running `npm install` in a circleci environment. This is with node `v8.1.0` and npm `v5.0.3`. There is a `package-lock.json` file present. The command `npm install` command works locally.

```
npm install

> husky@0.13.3 uninstall /home/ubuntu/frontend/node_modules/husky
> node ./bin/uninstall.js

module.js:487
    throw err;
    ^

Error: Cannot find module 'normalize-path'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/ubuntu/frontend/node_modules/husky/src/index.js:3:17)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
npm WARN lifecycle husky@0.13.3~uninstall: continuing anyway husky@0.13.3 uninstall: `node ./bin/uninstall.js`
npm WARN lifecycle Exit status 1

> husky@0.13.4 install /home/ubuntu/frontend/node_modules/husky
> node ./bin/install.js

module.js:487
    throw err;
    ^

Error: Cannot find module 'find-parent-dir'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/ubuntu/frontend/node_modules/husky/src/index.js:4:21)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
npm WARN frontend@0.205.0 license should be a valid SPDX license expression

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! husky@0.13.4 install: `node ./bin/install.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the husky@0.13.4 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ubuntu/.npm/_logs/2017-06-09T12_27_27_946Z-debug.log

npm install returned exit code 1

Action failed: npm install
```

**Comment by chrisui** [comment]
@typicode Do you think you will have chance to look at this soon? This is a blocker for us updating to node 8 (for whatever reason it's not working post update on our CI).

If you haven't got time right now I could take a look but would be great to have some pointers of where to start in the codebase! :)

**Comment by typicode** [comment]
@chrisui seems like when `npm` runs husky's `uninstall.js` script, husky's dependencies are already uninstalled and so it fails.

Not sure how to fix it though. I'll try to have a look tonight.

**Comment by typicode** [comment]
On your CI, maybe you can disable `npm` hooks:

```
npm install --ignore-scripts
```

This way it won't install husky, but it shouldn't be an issue since husky is a dev tool and not really meant to be run in a CI.

**Comment by chrisui** [comment]
@typicode This will interfere with all our other dependencies if they have script hooks, right? Will give it a go regardless. Thanks!

**Comment by typicode** [comment]
@chrisui yes, unfortunatly. There seem to be an issue on npm that may be related
https://github.com/npm/npm/issues/11894

**Comment by j4zz** [comment]
For us, this was fixed with *npm 5.5.1* and *husky 0.14.3*.
